inc/namespace.php

```diff
@@ -246,6 +262,109 @@ function noop_wp_query_found_rows_on_failed_ep_request( string $sql, WP_Query $q
 	return '';
 }
 
+/**
+ * Add default initial options and settings on install.
+ *
+ * @return void
+ */
+function on_wp_install() {
+	// This option is used to determine the index name for backwards compat.
+	set_index_version( 3 );
+}
+
+/**
+ * When an existing index is deleted set the index version so
+ *
+ * @param array $query The Elasticsearch query.
+ * @param string|null $type The remote request type.
+ * @return void
+ */
+function on_delete_index( $query, ?string $type ) {
+	if ( $type !== 'delete_index' ) {
+		return;
+	}
+	// Set the version to 3.
+	if ( get_index_version() === 2 ) {
+		set_index_version( 3 );
+	}
+}
+
+/**
+ * Set the index version for the current site.
+ *
+ * @param integer $version The version number.
+ * @return void
+ */
+function set_index_version( int $version ) {
+	update_option( 'altis_search_index_version', $version );
+}
+
+/**
+ * Get the index version for the current site. Defaults to
+ * 2 for ElasticPress version 2.
+ *
+ * @return int
+ */
+function get_index_version() : int {
+	return get_option( 'altis_search_index_version', null ) ?? 2;
+}
+
+/**
+ * ElasticPress adds the indexable object type to index names. We can maintain backwards
+ * compatibility by filtering the posts indexable index name to remove this type.
+ *
+ * @param string $index The index name.
+ * @return string
+ */
+function filter_index_name( string $index ) : string {
+	// Back compat for Altis v3 & ElasticPress 2.x
+	// Version 3 of ElasticPress introduces Indexables allowing for user
+	// and term search integration. The new index names follow the pattern
+	// <site>-<indexable>-<blog-id> instead of <site>-<blog-id>.
+	if ( get_index_version() === 2 && strpos( $index, '-post' ) !== false ) {
+		$old_index = str_replace( '-post', '', $index );
+		if ( Elasticsearch::factory()->index_exists( $old_index ) ) {
+			return $old_index;
+		}
```

Comment:
Should this update the index version if the index doesn't exist?

Reply:
Hmm maybe yeah. Might actually be better to remove this check, I should've commented at the time why I did it - it was to prevent an error when reindexing from the old name to the new one.